### PR TITLE
Merge #15765

### PR DESCRIPTION
**Pull request by FArthur-cmd**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Merge #15765 (Dynamic reload of server TLS certificates on config reload)
cc @johnskopis

Detailed description / Documentation draft:
Merge #15765

### CODE REVIEW

File: programs/server/config.xml [hunk 1]
```diff
@@ -237,14 +237,15 @@
     <openSSL>
         <server> <!-- Used for https server AND secure tcp port -->
             <!-- openssl req -subj "/CN=localhost" -new -newkey rsa:2048 -days 365 -nodes -x509 -keyout /etc/clickhouse-server/server.key -out /etc/clickhouse-server/server.crt -->
+            <!-- Uncomment certificateFile and privateKeyFile if you want to use them. -->
```
Comment (alexey-milovidov) [inline programs/server/config.xml:240]: But they're already uncommented.

File: programs/server/config.xml [hunk 2]
```diff
@@ -237,14 +237,15 @@
...
             <certificateFile>/etc/clickhouse-server/server.crt</certificateFile>
             <privateKeyFile>/etc/clickhouse-server/server.key</privateKeyFile>
             <!-- dhparams are optional. You can delete the <dhParamsFile> element.
                  To generate dhparams, use the following command:
                   openssl dhparam -out /etc/clickhouse-server/dhparam.pem 4096
                  Only file format with BEGIN DH PARAMETERS is supported.
               -->
-            <dhParamsFile>/etc/clickhouse-server/dhparam.pem</dhParamsFile>
+            <!-- <dhParamsFile>/etc/clickhouse-server/dhparam.pem</dhParamsFile> -->
```
Comment (alexey-milovidov) [inline programs/server/config.xml:248]: Maybe this change affected something? Looks unrelated to the PR.

Comment (FArthur-cmd) [reply programs/server/config.xml]: I was trying to prevent some server fails, but later found a better solution and forgot to undo these changes.

**Comment by FArthur-cmd** [comment]
02204_fractional_progress_bar is not related to this pr (There were some troubles in master that are already fixed)
